### PR TITLE
OCPBUGS-1871: Fix machine not going failed with invalid vmsize

### PR DESCRIPTION
**Pull request by RadekManak**
This error needs to be InvalidMachineConfiguration, because machine goes failed only if it finds error of that type.

### CODE REVIEW

**Comment by RadekManak** [comment]
/unhold
We treat failure to send request to create machine as invalid machine configuration, so I think failure to retrieve instance type information can be invalid machine configuration as well.

File: pkg/cloud/azure/actuators/machine/reconciler.go [hunk 1]
```diff
@@ -562,7 +562,7 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 
 		skuI, err := s.resourcesSkus.Get(ctx, skuSpec)
 		if err != nil {
-			return fmt.Errorf("failed to find sku %s", s.scope.MachineConfig.VMSize)
+			return machinecontroller.InvalidMachineConfiguration("failed to fetch instance type information from azure: %w", err)
```
Comment (JoelSpeed) [inline pkg/cloud/azure/actuators/machine/reconciler.go:565]: Could this be a transient error? What machine configuration leads into this? Is there a way to be more precise here, eg, if it returns a not found error vs something else?

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-azure-operator

The failure in the e2e is resolved by openshift/machine-api-operator#1076 and the tests that didn't run are autoscaler related, confident this change isn't going to break those

**Comment by JoelSpeed** [comment]
/hold

@RadekManak There are some comments in the bug about the message having changed, how would you like to resolve that?

**Comment by RadekManak** [comment]
Made it more clear that the request has not failed when the machine type is not found and added lists of valid VMsizes to the error.
/unhold

**Comment by JoelSpeed** [comment]
/hold

This failed QE, seems we need to do a similar change in another part of the code when we haven't got accelerated networking enabled

**Comment by RadekManak** [comment]
/retest
Did the tests get stuck?

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-azure-operator

This failed on the penultimate test which we are working on fixing, this is a known issue. I don't believe this PR is causing any problems. All the previous tests have passed and 2 of the 4 autoscaler tests passed indicating the autoscaler is also functional

**Pull request merged**